src/cpp/session/SessionConsoleProcess.cpp

```diff
@@ -251,6 +253,67 @@ void ConsoleProcess::commonInit()
                            module_context::activeSession().id());
 #endif
    }
+   
+   // if RStudio has been configured to use Python,
+   // place that copy of Python on the PATH
```

Comment:
Also, given the potential complexity of people's terminal shell configs, some might not want this happening at all, so I'd strongly recommend adding a terminal option to disable it. Probably in the Terminal / General / Shell section, and could be keyed to the selected shell (if they select a shell that doesn't support this feature maybe replace the checkbox with a message about Python settings not being automatically set).

Reply:
I've added an option to the Terminal preferences pane:

<img width="613" alt="Screen Shot 2021-04-21 at 12 02 06 PM" src="https://user-images.githubusercontent.com/1976582/115607105-669a0d00-a299-11eb-8d22-9bed17c77b24.png">

Does that look reasonable?

Reply:
Pending other discussion on WSL support (more details in later comments), might need to change the wording (a WSL user can have bash or wsl or some other shell inside the Linux environment, but I don't think this feature is going to work for any of those and the wording suggests it will).

Maybe have the checkbox text more closely match the shell options, so on Windows should say "(Git Bash only)?

Could perhaps hide the checkbox entirely when a shell is selected (Custom, PowerShell, Command Prompt, possibly WSL) that doesn't support this feature. In my mind it makes more sense for this checkbox to live under the Shell section rather than adding Miscellaneous, since it is only applicable to certain Shell settings (e.g. like the "Custom shell command-line options" control).